sdk/src/main/java/io/opentelemetry/sdk/metrics/BaseBoundInstrument.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.LabelSet;
+
+class BaseBoundInstrument {
+  private final LabelSet labels;
+
+  BaseBoundInstrument(LabelSet labels) {
+    this.labels = labels;
```

Comment:
I think labelset cannot be null, remove all the checks for null in the code?

Reply:
are you referring to the auto-generated equals/hashcode methods? I'd rather not mess with those if I don't have to. I don't think I see any other null checks.

Reply:
auto-generated by what?

Reply:
Going to merge this and will address this comment in a separate PR.

Reply:
it was auto-generated by IDEA, and I can have it do that with non-null as well. I'll put that PR in real quick